### PR TITLE
Add fusion into containing op.

### PR DESCRIPTION
**Pull request by nicolasvasilache**
- Alternative fusion variant that fuses a producer into a containing operation.
    
Co-authored-by: Tobias Gysi <gysit@users.noreply.github.com>

### CODE REVIEW

**Review by matthias-springer — APPROVED** [review]

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Fusion.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 The IREE Authors
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Fusion.cpp:1]: Nit: this should be 2022

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
This seems similar to the other PR. Is there a dependence here?

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Fusion.cpp (hunk 1 above)
Comment (nicolasvasilache) [reply llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Fusion.cpp:1]: > This seems similar to the other PR. Is there a dependence here?

This one is stacked on top of the other one and adds a second op with a different type of user control for fusion.

Comment (nicolasvasilache) [reply llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Fusion.cpp:1]: Fixed, thanks!

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**